pkg/apb/secrets.go

```diff
@@ -0,0 +1,159 @@
+package apb
+
+import (
+	"sync"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+)
+
+// SecretsConfig - Entry for a secret config block in broker config
+type SecretsConfig struct {
+	Title   string `yaml:"title"`
+	ApbName string `yaml:"apb_name"`
+	Secret  string `yaml:"secret"`
+}
+
+// Validate - Ensures that the secrets config is valid (ie, all strings are non-empty
+func (c SecretsConfig) Validate() bool {
+	for _, str := range []string{c.Title, c.ApbName, c.Secret} {
+		if str == "" {
+			return false
+		}
+	}
+	return true
+}
+
+// AssociationRule - A rule to associate apbs with a secrets
+type AssociationRule struct {
+	apbName string
+	secret  string
+}
+
+type secretsCache struct {
+	mapping map[string][]string
+	rwSync  sync.RWMutex
+	rules   []AssociationRule
+	config  []SecretsConfig
+	log     *logging.Logger
+}
+
+var secrets secretsCache
+
+// GetSecrets - Returns a list of secrets to be attached to a specified spec
+func GetSecrets(spec *Spec) []string {
+	secrets.rwSync.RLock()
+	defer secrets.rwSync.RUnlock()
+	return secrets.mapping[spec.FQName]
+}
```

Comment:
It does impact testing annoyingly, but I made it take the entire spec so that no other modules need to care about how we key the specs in the cache. If we change it to be keyed on something other than FQName in the next iteration (which I could see happening), we won't be changing the API to this module.